lib/net/ssh/authentication/certificate.rb

```diff
@@ -31,7 +31,10 @@ def self.read_certblob(buffer, type)
           cert.key_id = buffer.read_string
           cert.valid_principals = buffer.read_buffer.read_all(&:read_string)
           cert.valid_after = Time.at(buffer.read_int64)
-          cert.valid_before = Time.at(buffer.read_int64)
+          # 0x20c49ba5e353f7 = 0x7fffffffffffffff/1000, the largest value possible for JRuby
+          # JRuby Time.at multiplies the arg by 1000, and then stores it in a signed long.
+          # 0x20c49ba5e353f7 = 292278994-08-17 01:12:55 -0600
+          cert.valid_before = Time.at([0x20c49ba5e353f7, buffer.read_int64].min)
```

Comment:
Please put it to an if statement so only executed on jruby

Reply:
I debated if it was better to have consistent behavior on all platforms, or limit to just JRuby. Fixed.